src/test/java/htsjdk/variant/vcf/VCFPathReaderTest.java

```diff
@@ -0,0 +1,61 @@
+package htsjdk.variant.vcf;
+
+import htsjdk.HtsjdkTest;
+import htsjdk.samtools.util.IOUtil;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.nio.file.Paths;
+
+/**
+ * Created by farjoun on 10/12/17.
+ */
+public class VCFPathReaderTest extends HtsjdkTest {
+
+    @DataProvider(name = "pathsData")
+    Object[][] pathsData() {
+        return new Object[][]{
+                // various ways to refer to a local file
+                {"src/test/resources/htsjdk/variant/VCF4HeaderTest.vcf", false, true},
+
+                {Paths.get("").toAbsolutePath().toString() + "/src/test/resources/htsjdk/variant/VCF4HeaderTest.vcf", false, true},
+                {"file://" + Paths.get("").toAbsolutePath().toString() + "/src/test/resources/htsjdk/variant/VCF4HeaderTest.vcf", false, true},
+
+                //testing GCS files:
+
+                // this is almost a vcf, but not quite it's missing the #CHROM line and it has no content...
+                {"gs://htsjdk-testdata/htsjdk/variant/Homo_sapiens_assembly38.tile_db_header.vcf", false, false},
```

Comment:
Tests for other FS could use Jimfs: copy the local files into Jimfs, and use the filesystem for the test to create the strings...

Reply:
done